torch/__init__.py

```diff
@@ -167,9 +167,16 @@ def _load_dll_libraries() -> None:
         else:
             base_py_dll_path = ""
 
+        # when using pip install on a project that depends on PyTorch, pip installs
+        # torch & mkl in a custom path that's included in sys.path but does not match
+        # any of the other DLLs path above, so we need to manually add it here.
+        mkl_dll_path = os.path.join(
+            os.path.dirname(__file__), "..", "..", "..", "Library", "bin"
```

Comment:
This feels wrong, I would prefer it to be expressed as `sysconfig.get_config_var("XYZ")` + something else, because during the normal installation this will add a search path to some random folder, that could be used to mess with some existing installs

Reply:
I had a look through what's in `sysconfig.get_config_vars()`, but this was not here. I think this is really a specific of how pip does build isolation. Another solution might be to explicitly search for mkl distribution and use this path, but I could not find a way to find it other than walking through `sys.path` and looking for `mkl.dist-info` (`import mkl` fails since the package does not install any python files)